test/golden_tsickle_test.ts

```diff
@@ -183,17 +183,27 @@ testFn('golden tests with transformer', () => {
               Array.from(tsSources.keys())}`);
         }
       }
+      const isDeclarationTest = /\.declaration\b/.test(test.name);
       const {diagnostics, externs} = tsickle.emitWithTsickle(
           program, transformerHost, tsHost, tsCompilerOptions, targetSource,
           (fileName: string, data: string) => {
-            if (!fileName.endsWith(/\.declaration\b/.test(test.name) ? '.js' : '.d.ts')) {
-              // Normally we don't check .d.ts files, we are only interested to test that
-              // we don't throw when we generate them, but if we're in a .declaration test,
-              // we only care about the .d.ts files
-              tscOutput[fileName] = data;
+            if (isDeclarationTest) {
+              // Only compare .d.ts files for declaration tests.
+              if (!fileName.endsWith('.d.ts')) return;
+            } else {
+              // Only compare .js files for declaration tests.
```

Comment:
`// Only compare .js files for non-declaration tests.`
or
`// Only compare .js files for regular tests.`

Reply:
whoops, done.